### ISSUE TITLE
Adding Workspace/Projects page on Galaxy

### ISSUE THREAD

**Issue by pravs3683** [galaxyproject/galaxy#3269]
Hi Galaxy Team,

I was thinking about a feature for the Galaxy that will be helpful to many researchers. I wanted to propose something to organize/assign histories to different projects. The idea is to add one more tab at the top of the page, next to "Analyze Data" and "Workflow", named as "Workspace/Projects".
Clicking this link would list all the Projects that a user has worked on, and each of these projects will have selected/relevant histories listed there.

Please see the below image:

![galaxy_workspaceprojects](https://cloud.githubusercontent.com/assets/458683/20935431/a389a92c-bbd6-11e6-9837-759643be746c.png)

Thanks,
Praveen Kumar (kumar207@umn.edu, pravs3683@gmail.com)

Ph.D. Student,
Tim Griffin's Lab (Galaxy-P Team),
University of Minnesota,
Minneapolis, Minnesota

**Comment by bgruening** [comment]
Our users also requesting something similar regularly.

**Comment by jgoecks** [comment]
@pravs3683 @bgruening We suggest using tags to label and group histories. Do your use cases ask more than what tags can provide?

**Comment by jmchilton** [comment]
Yeah - some of this is definitely possible with tags. Maybe it is matter of easing the use of tags and making it more obvious:

![multihistory_view](https://cloud.githubusercontent.com/assets/216771/21506847/64771d8e-cc41-11e6-8c53-c3a86e43f44d.png)

For instance in the gray search bar one can imagine the most common tags appearing in a second row (maybe 5 most common with more available in a drop down) and when these are clicked the search box could add ``tag:<name>`` or whatever the search syntax is. Just seeing these there would encourage people to explore and use them.

The other idea would be when sending workflows to new histories based on input to have an option to link all of those histories together with a tag.

**Comment by jgoecks** [comment]
Agree completely that we need to do more with tags at the UI level. Building on @jmchilton's list:

- making it clear that tags are possible
- making it clear that searching by tags works
- showing common tags
- automatically grouping histories (and other objects) based on tags
- propagating tags when running tools and/or workflows

I imagine this work may be interesting to @guerler or @bgruening's UI folks.

**Comment by pravs3683** [comment]
Tag are good to work with. It really works the way I was thinking. It would
be helpful to have a "tags" tab on the top menu, next to "Analyze". That
would make it obvious that this feature is present. Also, easy to select
tags.



Praveen Kumar

Contact: +1 612-598-7760

**Comment by bgruening** [comment]
We are experimenting with something related to this issue in this branch: https://github.com/bgruening/galaxy/tree/tool_search_overlay

It was intended as an overlay search, to pin search results (like tools, workflows) and remove others. Like users are not using certain tools can remove them, other tools that they use most, can be pinned ...
We are using the entire screen as an overlay and have a feature like 10 most used tools. As a next step we would like to include a own category for tags, and give an overview about them, group them, pin/remove tags, etc ...

Everything is implemented as Galaxy Webhook and is completely optional, as soon as this is merged: https://github.com/galaxyproject/galaxy/pull/3349

Plan is to polish what we have currently, create a PR to gather more comments. if this gets merged at some point we would like to work on the tagging part. Any comments, review, testing is highly appreciated :)

**Comment by dannon** [comment]
This is probably good for now, but it might be worth looking into unifying masthead access down the road.

**Comment by hexylena** [comment]
This came up at lunch with @joachimwolff  /@bgruening recently, maybe overloading hashtags would be good here? They have that "high visibility" in that they appear even when tags aren't shown on datasets, we could do the same for histories?
